### PR TITLE
Add --docker-host flag to pass DOCKER_HOST env var to calico node

### LINKED ISSUE #1404: Pass DOCKER_HOST to calico/node
See https://github.com/projectcalico/libnetwork-plugin/issues/90#issuecomment-266078037 for detail

### PR DESCRIPTION
**Pull request by gunjan5**
Fixes #1404

### CODE REVIEW

File: calicoctl/commands/node/run.go [hunk 1]
```diff
@@ -213,9 +217,10 @@ Description:
...
 	}
 
-	// Set CALICO_LIBNETWORK_IFPREFIX env variable if Docker network is enabled.
+	// Set DOCKER_HOST and CALICO_LIBNETWORK_IFPREFIX env variables if Docker network is enabled.
 	if !disableDockerNw {
 		envs["CALICO_LIBNETWORK_IFPREFIX"] = ifprefix
+		envs["DOCKER_HOST"] = dockerhost
 	}
 
 	// Add in optional environments.
```
Comment (robbrockbank) [inline calicoctl/commands/node/run.go]: See line 247 -  you also need to change your volume mounts

**Comment by gunjan5** [comment]
@robbrockbank not sure what I would have to mount if the docker host isn't the default local unix socket. It would be something like this `DOCKER_HOST=tcp://172.17.8.101:2376` what would I mount in this case?

**Comment by robbrockbank** [comment]
I assume for tcp://  format we don't need to volume mount in the socket.  But for unix:// we need to volume mount in the socket to the expected location.

Probably worthy of a separate issue, we should consider adding the ability to specify certificates used for TLS authentication on the Docker TCP connection (see DOCKER_CERT_PATH).

**Comment by robbrockbank** [comment]
Sorry pressed the wrong button and accidentally closed.  Reopened now :-)

**Comment by tomdee** [comment]
@gunjan5 @robbrockbank What's happening with this PR?

**Comment by gunjan5** [comment]
closing this for now.

**Pull request closed without merging**